continuous_integration/home/coder/.local/bin/entrypoint

```diff
@@ -1,7 +1,26 @@
 #!/usr/bin/env bash
 
-entrypoint() {
+sccache_stop_server_and_show_stats() {
+    sccache --stop-server || true && sccache --show-stats;
+}
+
+init_devcontainer() {
+    # disable xtrace and history
+    local xtrace_enabled=$(echo "${SHELLOPTS:-}" | grep -q 'xtrace'; echo $?);
+    local history_enabled=$(echo "${SHELLOPTS:-}" | grep -q 'history'; echo $?);
+    { set +xo history; } 2>/dev/null;
+    eval "export $(find /run/secrets/ -type f -exec bash -c 'echo ${0/\/run\/secrets\//}=$(<${0})' {} \;)";
+    if [ "${history_enabled}" -eq "0" ]; then { set -o history; } 2>/dev/null; fi;
+    if [ "${xtrace_enabled}" -eq "0" ]; then { set -o xtrace; } 2>/dev/null; fi;
+
+    . devcontainer-utils-post-attach-command;
 
+    sleep 10;
+    . devcontainer-utils-vault-s3-test;
+    . devcontainer-utils-vault-s3-export 0;
+}
```

Comment:
My understanding of this is lacking. :) Why do we need this?

Reply:
> devcontainer-utils-vault-s3-test

This tests if AWS credentials are valid.

> devcontainer-utils-vault-s3-export

This sets up various SCCACHE_* env vars through ~/.aws/config or ~/.bashrc.

This things are required for enabling SCCACHE for "local repro" where AWS auth happens using a GH personal access token.

Reply:
I would also add that the 10 second wait is essential. The credentials that we get are generated on the fly, and need to be uploaded to AWS. We hit authentication issues if we try to race against the upload process.

Reply:
This is also done in the 23.08 devcontainers (and removed in https://github.com/nv-legate/legate.core/pull/725).